src/t8_schemes/t8_new_feature/t8_subelements/t8_subelements_quad_cxx.cxx

```diff
@@ -0,0 +1,1325 @@
+   * This is the case if the child_id matches one of the faces corners */
+  child_id = p4est_quadrant_child_id (q);
+  if (child_id == p4est_face_corners[face][0]
+      || child_id == p4est_face_corners[face][1]) {
+    return face;
+  }
+  return -1;
+}
+
+void
+t8_default_scheme_sub_c::t8_element_transform_face (const t8_element_t *
+                                                     elem1,
+                                                     t8_element_t * elem2,
+                                                     int orientation,
+                                                     int sign,
+                                                     int is_smaller_face)
+{
+  const t8_quad_with_subelements *pquad_w_sub_elem1 = (const t8_quad_with_subelements *) elem1;
+  t8_quad_with_subelements *pquad_w_sub_elem2 = (t8_quad_with_subelements *) elem2;
+
+  const p4est_quadrant_t *qin = &pquad_w_sub_elem1->p4q;
+  p4est_quadrant_t *p = &pquad_w_sub_elem2->p4q;
+
+  const p4est_quadrant_t *q;
+  p4est_qcoord_t      h = P4EST_QUADRANT_LEN (qin->level);
+  p4est_qcoord_t      x = qin->x;       /* temp storage for x coordinate in case elem1 = elem 2 */
+
+  T8_ASSERT (t8_element_is_valid (elem1));
+  T8_ASSERT (t8_element_is_valid (elem2));
+  T8_ASSERT (0 <= orientation && orientation < P4EST_FACES);
+
+  if (sign) {
+    /* The tree faces have the same topological orientation, and
+     * thus we have to perform a coordinate switch. */
+    /* We use p as storage, since elem1 and elem2 are allowed to
+     * point to the same quad */
+    q = (const p4est_quadrant_t *) p;
+    t8_element_copy_surround (qin, (p4est_quadrant_t *) q);
+    ((p4est_quadrant_t *) q)->x = qin->y;
+    ((p4est_quadrant_t *) q)->y = x;
+    x = q->x;                   /* temp storage in case elem1 = elem 2 */
+  }
+  else {
+    q = qin;
+  }
+
+  p->level = q->level;
+  /*
+   * The faces of the root quadrant are enumerated like this:
+   *
+   *   v_2      v_3
+   *     x -->-- x
+   *     |       |
+   *     ^       ^
+   *     |       |
+   *     x -->-- x
+   *   v_0      v_1
+   *
+   * Orientation is the corner number of the bigger face that coincides
+   * with the corner v_0 of the smaller face.
+   */
+  /* If this face is not smaller, switch the orientation:
+   *  sign = 0   sign = 1
+   *  0 -> 0     0 -> 0
+   *  1 -> 2     1 -> 1
+   *  2 -> 1     2 -> 2
+   *  3 -> 3     3 -> 3
+   */
+  if (!is_smaller_face && (orientation == 1 || orientation == 2) && !sign) {
+    orientation = 3 - orientation;
+  }
+
+  switch (orientation) {
+  case 0:                      /* Nothing to do */
+    p->x = q->x;
+    p->y = q->y;
+    break;
+  case 1:
+    p->x = P4EST_ROOT_LEN - q->y - h;
+    p->y = x;
+    break;
+  case 2:
+    p->x = q->y;
+    p->y = P4EST_ROOT_LEN - x - h;
+    break;
+  case 3:
+    p->x = P4EST_ROOT_LEN - q->x - h;
+    p->y = P4EST_ROOT_LEN - q->y - h;
+    break;
+  default:
+    SC_ABORT_NOT_REACHED ();
+  }
+  T8_QUAD_SET_TDIM (p, 2);
```

Comment:
Setting subelement infos shouldnt be necessary since elem2 is initialized with default values at this point and elem1 is not allowed to be a subelement (at this point)? Might change in the future.